pkg/operator/controllers/etchosts/cluster_controller.go

```diff
@@ -68,43 +89,47 @@ func (r *EtcHostsClusterReconciler) Reconcile(ctx context.Context, request ctrl.
 	// If 99-master-aro-etc-hosts-gateway-domains doesn't exist, create it
 	err = r.Client.Get(ctx, types.NamespacedName{Namespace: "openshift-machine-api", Name: "99-master-aro-etc-hosts-gateway-domains"}, mc)
 	if kerrors.IsNotFound(err) {
-		err = r.Client.Get(ctx, types.NamespacedName{Name: "master"}, mcp)
+		err = r.Client.Get(ctx, types.NamespacedName{Name: "99-master-aro-etc-hosts-gateway-domains"}, mc)
 		if kerrors.IsNotFound(err) {
 			r.ClearDegraded(ctx)
-			return reconcile.Result{}, nil
-		}
-		if err != nil {
-			r.Log.Error(err)
-			r.SetDegraded(ctx, err)
+			r.Log.Debug("machine config 99-master-aro-etc-hosts-gateway-domains not found, creating it")
+			err = reconcileMachineConfigs(ctx, instance, "master", r.dh, *mcp)
+			if err != nil {
+				r.Log.Error(err)
+				r.SetDegraded(ctx, err)
+				return reconcile.Result{}, err
+			}
 			return reconcile.Result{}, err
 		}
-		err = reconcileMachineConfigs(ctx, instance, "master", r.dh, *mcp)
 		if err != nil {
 			r.Log.Error(err)
 			r.SetDegraded(ctx, err)
 			return reconcile.Result{}, err
 		}
+		return reconcile.Result{}, nil
```

Comment:
IMO we should requeue here

```suggestion
		return reconcile.Result{Requeue: true}, nil
```

Reply:
marking unresolved - assuming this got force-pushed over after `Accept Suggestion` was clicked ;)